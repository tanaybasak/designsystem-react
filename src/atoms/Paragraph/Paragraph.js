import React from 'react';
import PropTypes from 'prop-types';

const Paragraph = ({ className, children, ...restProps }) => {
  return (
    <p className={className} {...restProps}>
      {children}
    </p>
  );
};

Paragraph.propTypes = {
  /** Paragraph Content */
  children: PropTypes.node,
  /** Style class of the component */
  className: PropTypes.string
};

Paragraph.defaultProps = {
  children: null,
  className: ''
};

export default Paragraph;
