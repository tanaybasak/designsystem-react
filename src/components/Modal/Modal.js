import React from 'react';
import PropTypes from 'prop-types';
import './Modal.scss';

const Modal = props => (
    <React.Fragment>
        <div className="patron-modal">
            <div className="patron-modal-dialog">
                <div className="patron-modal-content">
                    <div className="patron-modal-header">
                        <h4>{props.title}</h4>
                        <button
                            type="button"
                            className="patron-modal-close"
                            data-btn-type="cancel"
                            onClick={props.onClick}
                        />
                    </div>
                    <div className="patron-modal-body">
                        <props.render />
                    </div>
                    {props.footer && props.footer.length ?
                        <div className="patron-modal-footer">
                            {props.footer.map((item, index) => (
                                <button
                                    className={`patron-btn patron-btn-${item.type}`}
                                    data-btn-type={item.btnType}
                                    key={`${item.btnType}-${index}`}
                                    onClick={props.onClick}
                                >
                                    {item.label}
                                </button>
                            ))}
                        </div>
                        : null
                    }
                </div>
            </div>
        </div>
        <div className="patron-modal-backdrop"></div>
    </React.Fragment>
);

Modal.defaultProps = {
    title: 'Modal Title',
    onClick: event => { event.preventDefault(); },
    footer: [],
    render: (() => <div>Body</div>)(),
};

Modal.propTypes = {
    title: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
    footer: PropTypes.array,
    render: PropTypes.func.isRequired,
};

export default Modal;