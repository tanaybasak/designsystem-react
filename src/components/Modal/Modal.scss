.patron-modal {
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: 1050;
    display: block;
    overflow: hidden;
    outline: 1;
    z-index: 1072;
    transition: opacity .15s linear;
    overflow-x: hidden;
    overflow-y: auto;
    padding-right: 16px;
    .patron-modal-dialog {
        transition: -webkit-transform .3s ease-out;
        transition: transform .3s ease-out;
        transition: transform .3s ease-out, -webkit-transform .3s ease-out;
        -webkit-transform: translate(0, 0);
        transform: translate(0, 0);
        max-width: 500px;
        margin: 1.75rem auto;
        .patron-modal-content {
            position: relative;
            display: -webkit-box;
            display: -ms-flexbox;
            display: flex;
            -webkit-box-orient: vertical;
            -webkit-box-direction: normal;
            -ms-flex-direction: column;
            flex-direction: column;
            width: 100%;
            pointer-events: auto;
            background-color: #fff;
            background-clip: padding-box;
            outline: 0;
            .patron-modal-header {
                padding: 0.5rem;
                border-bottom: 1px solid #e5e5e5;
                background-color: #047cc0;
                color: #fff;
                display: flex;
                align-items: flex-start;
                >h5 {
                    margin: 0;
                    font-weight: 400;
                }
                button.patron-modal-close {
                    background-color: transparent;
                    border: none;
                    outline: none;
                    cursor: pointer;
                    color: #fff;
                    font-weight: 400; 
                    font-size: 1.25rem;
                    margin: -0.5rem 0rem -1rem auto;
                    &::after {
                        content: 'x'
                    }
                }
            }
            .patron-modal-body {
                padding: 1rem;
            }
            .patron-modal-footer {
                padding: 0.825rem;
                text-align: right;
                border-top: 1px solid #e5e5e5;
            }
        }
    }
}

.patron-modal-backdrop {
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: 1040;
    background-color: #000;
    opacity: .5;
    z-index: 1071;
}

.patron-btn {
    font-size: 16px;
    outline: none;
    min-height: 35px;
    min-width: 70px;
    cursor: pointer;
    margin: 0 0.5rem;
    &-submit {
        background-color: #40d5bb;
        outline: none;
        border: 1px solid #40d5bb;
        color: white;
        &:hover {
            background-color: #40d5bb;
        }
    }
    &-cancel {
        background-color: #fff;
        border: 1px solid #343a40;
        color: #000;
    }
}