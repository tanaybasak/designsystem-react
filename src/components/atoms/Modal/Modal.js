import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { prefix } from '../../../settings';

export default function Modal({ type, label, heading, content, footer }) {

    return (
        <section className="hcl-modal hcl-modal-hide">
            <div className={`hcl-modal-container hcl-modal-container-lg  ${type === 'danger' ? 'hcl-modal-container-danger ' : null}`}>
                <header className="hcl-modal-header">
                    {label !== '' ? <small className="hcl-modal-label">{label}</small> : null}
                    <h5>{heading}</h5>
                </header>
                <div className="hcl-modal-content">
                    {content !== '' ? <p> {content}</p> : null}
                </div>
                {footer ?
                    <footer className="hcl-modal-footer ">
                        <button id="hcl-modal-cancel-type3" className="hcl-btn hcl-primary hcl-primary-outline">Cancel</button>

                        {type === 'danger' ?
                            <button class="hcl-btn hcl-danger">Delete</button>
                            :
                            <button className="hcl-btn hcl-primary">Save</button>
                        }
                    </footer>
                    : null
                }
                <button id="hcl-modal-close-type1" className="hcl-modal-close"></button>
            </div>
        </section>
    );
};

Modal.propTypes = {
    type: PropTypes.string,
    label: PropTypes.string,
    heading: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
    footer: PropTypes.bool,
};

Modal.defaultProps = {
    type: 'default',
    label: '',
    // heading: 'heading',
    // content: 'This is temporary content',
    footer: false,
};