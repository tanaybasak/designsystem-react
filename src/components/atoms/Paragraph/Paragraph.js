import React from "react";
import PropTypes from "prop-types";

export default function Paragraph({ className, children, ...restProps }) {
  return (
    <p className={className} {...restProps}>
      {children}
    </p>
  );
}

Paragraph.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string
};

Paragraph.defaultProps = {
  children: null,
  className: ""
};
