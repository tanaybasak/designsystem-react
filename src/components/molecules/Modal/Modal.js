import React from 'react';
import PropTypes from 'prop-types';
import prefix from '../../../settings';
import ActionBar from '../../atoms/ActionBar';

const Modal = ({
  type,
  label,
  heading,
  children,
  onClose,
  actions
}) => {
  const classNames = [`${prefix}-modal-container ${prefix}-modal-container-lg`];

  if(type === 'danger')
    {
      classNames.push(`${prefix}-modal-container-danger`);
    }

  return (
    <section className={`${prefix}-modal`}>
      <div className={classNames.join(' ')}>
        <button
          type='button'
          className={`${prefix}-modal-close`}
          onClick={onClose}
        />
        {(heading !== '' || label !== '') && (
          <header className={`${prefix}-modal-header`}>
            {label !== '' ? (
              <small className={`${prefix}-modal-label`}>{label}</small>
            ) : null}
            {heading !== '' && <h5>{heading}</h5>}
          </header>
        )}
        {children && (
          <div className={`${prefix}-modal-content`}>{children}</div>
        )}
        {actions.length > 0 && (
          <footer className={`${prefix}-modal-footer`}>
            <ActionBar actions={actions} />
          </footer>
        )}
      </div>
    </section>
  );
}

Modal.propTypes = {
  type: PropTypes.string,
  label: PropTypes.string,
  heading: PropTypes.string,
  onClose: PropTypes.func,
  actions: PropTypes.arrayOf([]),
  children: PropTypes.node.isRequired
};

Modal.defaultProps = {
  type: 'default',
  label: '',
  heading: '',
  onClose: () => {},
  actions: []
};

export default Modal;