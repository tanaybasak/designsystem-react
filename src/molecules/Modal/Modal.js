import React, { useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import Actions from '../../atoms/Actions';
import prefix from '../../settings';
import { Close } from '../../util/icons';
const Modal = ({
  type,
  label,
  heading,
  children,
  showClose,
  onClose,
  actions,
  keyboard,
  className,
  ...restProps
}) => {
  const modal = useRef(null);
  const classNames = [
    `${prefix}-modal-container ${prefix}-modal-container-lg ${className}`
  ];

  if (type === 'danger') {
    classNames.push(`${prefix}-modal-container-danger`);
  }

  useEffect(() => {
    modal.current.focus();
  }, []);

  const focusTrap = e => {
    const focusableEls = modal.current.querySelectorAll(
      'a[href]:not([disabled]), button:not([disabled]), textarea:not([disabled]), input[type="text"]:not([disabled]), input[type="radio"]:not([disabled]), input[type="checkbox"]:not([disabled]), select:not([disabled]), [tabindex]'
    );
    const firstFocusableEl = focusableEls[0];
    const lastFocusableEl = focusableEls[focusableEls.length - 1];

    if (e.keyCode == 27) {
      if (keyboard) {
        e.preventDefault();
        onClose();
      }
    }

    const isTabPressed = e.key === 'Tab' || e.keyCode === '9';

    if (!isTabPressed) {
      return;
    }

    if (e.shiftKey) {
      if (document.activeElement === firstFocusableEl) {
        lastFocusableEl.focus();
        e.preventDefault();
      }
    } else {
      if (document.activeElement === lastFocusableEl) {
        firstFocusableEl.focus();
        e.preventDefault();
      }
    }
  };

  return (
    <section
      className={`${prefix}-modal`}
      {...restProps}
      tabIndex="0"
      ref={modal}
      onKeyDown={focusTrap}
    >
      <div className={classNames.join(' ')}>
        {showClose ? (
          <button
            type="button"
            className={`${prefix}-modal-close`}
            aria-label="modal-close"
            onClick={onClose}
          >
            {Close}
          </button>
        ) : null}
        {(heading !== '' || label !== '') && (
          <header
            className={`${prefix}-modal-header ${prefix}-modal-header-lg`}
          >
            {label !== '' ? (
              <small className={`${prefix}-modal-label`}>{label}</small>
            ) : null}
            {heading !== '' && <h5>{heading}</h5>}
          </header>
        )}
        {children && (
          <div className={`${prefix}-modal-content ${prefix}-modal-content-lg`}>
            {children}
          </div>
        )}
        {actions.length > 0 && (
          <footer
            className={`${prefix}-modal-footer ${prefix}-modal-footer-lg`}
          >
            <Actions actions={actions} />
          </footer>
        )}
      </div>
    </section>
  );
};

Modal.propTypes = {
  /** Default: To create success modal. 
Danger: : To create danger modal. */
  type: PropTypes.string,

  /** To create label of the modal. */
  label: PropTypes.string,

  /** To create heading of the modal. */
  heading: PropTypes.string,

  /** To toggle close button of the modal. */
  showClose: PropTypes.bool,

  /** A callback function which will be executed once modal is closed. */
  onClose: PropTypes.func,

  /** To create action items associated with modal. */
  actions: PropTypes.array,

  /** To pass content to modal. */
  children: PropTypes.node.isRequired,

  /** Class/clasess will be applied on the parent div of Modal */
  className: PropTypes.string,
  /** Closes the modal when escape key is pressed */
  keyboard: PropTypes.bool
};

Modal.defaultProps = {
  type: 'default',
  label: '',
  heading: '',
  showClose: true,
  onClose: () => {},
  actions: [],
  className: '',
  keyboard: true
};

export default Modal;
