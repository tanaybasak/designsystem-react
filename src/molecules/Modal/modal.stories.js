import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { select, text, object, boolean } from '@storybook/addon-knobs';
//@update-path-build-start
import Modal from './Modal';
//@update-path-build-end

const typeOptions = {
  Danger: 'danger',
  Default: 'default',
  Warning: 'warning',
  Ghost: 'ghost'
};

const actions = [
  { label: 'Save', primary: true, disabled: true },
  { label: 'Close', secondary: false }
];

storiesOf('Modal', module).add(
  'default',
  () => (
    <Modal
      actions={object('Actions', actions)}
      heading={text('Heading', 'Heading')}
      label={text('Label', 'label')}
      showClose={boolean('showClose', true)}
      onClose={action('Modal on Close')}
      type={select('Type', typeOptions, 'danger')}
    >
      <p>This is a new modal</p>
    </Modal>
  ),
  {
    info: {
      text: `Description About Modal Component \n
      import { Modal } from '@patron/patron-react/modal';`
    }
  }
);
