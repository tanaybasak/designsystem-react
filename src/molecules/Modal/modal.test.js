import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import prefix from '../../settings';
import Modal from './Modal';
import Paragraph from '../../atoms/Paragraph';

describe('<Modal> component', () => {
    it('snapshots/renders Modal correctly', () => {
        const tree = renderer.create(<Modal><Paragraph>Danger Modal with save and close buttons</Paragraph></Modal>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('heading & label given in Modal is rendered', () => {
        const modal = mount(<Modal heading="Heading comes here." label="optional label"><Paragraph>Danger Modal with save and close buttons</Paragraph></Modal>);
        expect(modal.exists(`.${prefix}-modal-header h5`)).toEqual(true);
        expect(modal.exists(`small.${prefix}-modal-label`)).toEqual(true);
    });

    it('actions gets called in modal', () => {
        const modalActions1 = [
            { label: "Save" },
            {
                label: "Close",
                handler: () => {
                    this.onModalClose();
                },
                danger: true
            }
        ];
        const modal = mount(<Modal actions={modalActions1}><Paragraph>Danger Modal with save and close buttons</Paragraph></Modal>);
        expect(modal.find(`footer.${prefix}-modal-footer`)).toBeTruthy();
    });
});