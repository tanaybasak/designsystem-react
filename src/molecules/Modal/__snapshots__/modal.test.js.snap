// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal> component snapshots/renders Modal correctly 1`] = `
<section
  className="hcl-modal"
  onKeyDown={[Function]}
  tabIndex="0"
>
  <div
    className="hcl-modal-container hcl-modal-container-lg "
  >
    <button
      aria-label="modal-close"
      className="hcl-modal-close"
      onClick={[Function]}
      type="button"
    />
    <div
      className="hcl-modal-content hcl-modal-content-lg"
    >
      <p
        className=""
      >
        Danger Modal with save and close buttons
      </p>
    </div>
  </div>
</section>
`;
